From 71cc9cbfe1b7330ef3685f78bf2e24318b15f04a Mon Sep 17 00:00:00 2001
From: "Travis F. Collins" <travis.collins@analog.com>
Date: Wed, 29 Jan 2020 16:40:33 -0800
Subject: [PATCH 3/9] Add pypi to travis-ci providers for python bindings. Only
 the first x86 Linux build will push to pypi on tagged commits.

Signed-off-by: Travis F. Collins <travis.collins@analog.com>
---
 .travis.yml | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/.travis.yml b/.travis.yml
index e786842..baf9ddb 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -35,6 +35,7 @@ matrix:
         # TODO do `check` or `check_hw` here too, not just `check_sw` (as part of `coverage`)
         - CHECK_RULE=coverage GCOV=1
         - PKG_RULE=gzip
+        - PYPI=yes
         - COVERITY_SCAN_PROJECT_NAME="$TRAVIS_REPO_SLUG"
         - COVERITY_SCAN_BRANCH_PATTERN="master"
         - COVERITY_SCAN_NOTIFICATION_EMAIL="robin.getz@analog.com"
@@ -114,6 +115,7 @@ before_deploy:
   - eval "$(ssh-agent -s)"
   - chmod 600 /tmp/deploy.rsa
   - ssh-add /tmp/deploy.rsa
+  - if [[ "$PYPI" == "yes" ]]; then cd "build/bindings/python"; fi
 deploy:
   - provider: releases
     api_key:
@@ -152,6 +154,16 @@ deploy:
     on:
       condition: "$TRAVIS_OS_NAME = osx"
       all_branches: true
+  - provider: pypi
+    if: tag IS present
+    skip_cleanup: true
+    on:
+      tags: true
+      condition: "$PYPI = yes"
+    username: __token__
+    password:
+      secure: KfORvnPyLlMANLpcwJjcO44Kt6meP614omJk2GloD+B51SuV64O6N31VDOTi2xqioEk9Cajq7tmNte5YoHVUjL9iWFsBblpzxZoXcy40dn4pFN0kMiE8scszxckbDwuBZhokUVZ2fhfLKf/8xORcwjS3vKqSj5HwQKKvVgukSbQ=
+    distributions: sdist
 
 after_deploy:
   - ${TRAVIS_BUILD_DIR}/CI/travis/after_deploy
-- 
2.20.1

